pg_shard--1.1--1.2.sql

```diff
@@ -1,13 +1,242 @@
+		ALTER SEQUENCE pgs_distribution_metadata.shard_placement_id_sequence
+			OWNED BY pgs_distribution_metadata.shard_placement.id;
+	END IF;
+END;
+$$;
+
+-- Syncs rows from the pg_shard distribution metadata related to the specified
+-- table name into the metadata tables used by CitusDB. After a call to this
+-- function for a particular pg_shard table, that table will become usable for
+-- queries within CitusDB. If placement health has changed for given pg_shard
+-- table, calling this function an additional time will propagate those health
+-- changes to the CitusDB metadata tables.
+CREATE OR REPLACE FUNCTION sync_table_metadata_to_citus(table_name text)
+RETURNS void
+AS $sync_table_metadata_to_citus$
+	DECLARE
+		table_relation_id CONSTANT oid NOT NULL := table_name::regclass::oid;
+		dummy_shard_length CONSTANT bigint := 0;
+		warning_msg CONSTANT text := 'sync_table_metadata_to_citus is deprecated and ' ||
+									 'will be removed in a future version';
+	BEGIN
+		RAISE WARNING '%', warning_msg;
+
+		-- grab lock to ensure single writer for upsert
+		LOCK TABLE pg_dist_shard_placement IN EXCLUSIVE MODE;
+
+		-- First, update the health of shard placement rows already copied
+		-- from pg_shard to CitusDB. Health is the only mutable attribute,
+		-- so it is presently the only one needing the UPDATE treatment.
+		UPDATE pg_dist_shard_placement
+		SET    shardstate = shard_placement.shard_state
+		FROM   pgs_distribution_metadata.shard_placement
+		WHERE  shardid = shard_placement.shard_id AND
+			   nodename = shard_placement.node_name AND
+			   nodeport = shard_placement.node_port AND
+			   shardid IN (SELECT shardid
+						   FROM   pg_dist_shard
+						   WHERE  logicalrelid = table_relation_id);
+
+		-- copy pg_shard placement rows not yet in CitusDB's metadata tables
+		INSERT INTO pg_dist_shard_placement
+					(shardid,
+					 shardstate,
+					 shardlength,
+					 nodename,
+					 nodeport)
+		SELECT shard_id,
+			   shard_state,
+			   dummy_shard_length,
+			   node_name,
+			   node_port
+		FROM   pgs_distribution_metadata.shard_placement
+			   LEFT OUTER JOIN pg_dist_shard_placement
+							ON ( shardid = shard_placement.shard_id AND
+								 nodename = shard_placement.node_name AND
+								 nodeport = shard_placement.node_port )
+		WHERE  shardid IS NULL AND
+			   shard_id IN (SELECT id
+							FROM   pgs_distribution_metadata.shard
+							WHERE  relation_id = table_relation_id);
+
+		-- copy pg_shard shard rows not yet in CitusDB's metadata tables
+		INSERT INTO pg_dist_shard
+					(shardid,
+					 logicalrelid,
+					 shardstorage,
+					 shardminvalue,
+					 shardmaxvalue)
+		SELECT id,
+			   relation_id,
+			   storage,
+			   min_value,
+			   max_value
+		FROM   pgs_distribution_metadata.shard
+			   LEFT OUTER JOIN pg_dist_shard
+							ON ( shardid = shard.id )
+		WHERE  shardid IS NULL AND
+			   relation_id = table_relation_id;
+
+		-- Finally, copy pg_shard partition rows not yet in CitusDB's metadata
```

Comment:
For the given relationId, there should be just one row right? The comment and the query are a bit confusing. Can we modify the query to denote that we expect only one row in this table for the given relationId?